### PR TITLE
Add tests, SUP-6451

### PR DESCRIPTION
**Pull request by BorisovskiP**

### CODE REVIEW

File: src/Test/Integration/Model/Content/Generator/PostTest.php [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        // @phpstan-ignore-next-line
+        self::$projectId = (int)$project->getId();
+    }
+
+    public static function loadPostsFixture(): void
+    {
+        include __DIR__ . '/../../../_files/two_posts_with_same_url_for_different_stores.php';
+    }
+
+    private function loadPostPost(string $urlKey, int $storeId): Post
```
Comment (sprankhub) [inline src/Test/Integration/Model/Content/Generator/PostTest.php:148]: Please rename :)

File: src/Test/Integration/Model/Content/Generator/PostTest.php [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+use EasyTranslate\Connector\Model\Project;
+use Magento\TestFramework\Helper\Bootstrap;
+use Mageplaza\Blog\Api\Data\PostInterface;
+use Mageplaza\Blog\Model\Post;
+use Mageplaza\Blog\Model\PostFactory;
+use Mageplaza\Blog\Model\ResourceModel\Post as PostResource;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @magentoDbIsolation  enabled
```
Comment (sprankhub) [inline src/Test/Integration/Model/Content/Generator/PostTest.php:20]: Why is that needed?

File: src/Test/Integration/Model/Content/Generator/PostTest.php [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+        $this->postsGenerator     = $objectManager->create(PostsGenerator::class);
+        $this->postFactory        = $objectManager->create(PostFactory::class);
+        $this->postResource       = $objectManager->create(PostResource::class);
+        $this->compatPostResource = $objectManager->create(CompatResourcePost::class);
+    }
+
+    /**
+     * @magentoDataFixture    loadPostsFixture
+     * @magentoDataFixture    loadProjectFixture
+     * @magentoAppIsolation   enabled
```
Comment (sprankhub) [inline src/Test/Integration/Model/Content/Generator/PostTest.php:67]: Why is that needed?

File: src/Test/Integration/Model/Content/Generator/PostTest.php [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+        $this->assertContent($identifier, $storeId, $includedAttributes, $excludedAttributes);
+    }
+
+    private function assertContent(
+        string $urlKey,
+        int $storeId,
+        array $includedAttributes,
+        array $excludedAttributes
+    ): void {
+
```
Comment (sprankhub) [inline src/Test/Integration/Model/Content/Generator/PostTest.php:111]: Please remove this line.

File: src/Test/Integration/_files/two_posts_with_same_url_for_different_stores.php [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+declare(strict_types=1);
+
+use Magento\Store\Api\StoreRepositoryInterface;
+use Magento\TestFramework\Helper\Bootstrap;
+use Magento\TestFramework\Workaround\Override\Fixture\Resolver;
+use Mageplaza\Blog\Model\ResourceModel\Post as PostResource;
+use Mageplaza\Blog\Model\Post;
+
+Resolver::getInstance()->requireDataFixture('Magento/Store/_files/second_store.php');
```
Comment (sprankhub) [inline src/Test/Integration/_files/two_posts_with_same_url_for_different_stores.php:11]: Please fix, so that it works in 2.3 as well.

File: src/Test/Integration/_files/two_posts_with_same_url_for_different_stores.php [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+/** @var StoreRepositoryInterface $storeRepository */
+$storeRepository = $objectManager->get(StoreRepositoryInterface::class);
+$store           = $storeRepository->get('fixture_second_store');
+
+/** @var PostResource $postResource */
+$postResource = $objectManager->create(PostResource::class);
+
+/** @var Post $post */
+$post = $objectManager->create(Post::class);
+$post->setName('Test1')
```
Comment (sprankhub) [inline src/Test/Integration/_files/two_posts_with_same_url_for_different_stores.php:24]: Please create a rollback file.

**Review by sprankhub — CHANGES_REQUESTED** [review]

File: src/Test/Integration/Model/Content/Generator/PostTest.php (hunk 3 above)
Comment (BorisovskiP) [reply src/Test/Integration/Model/Content/Generator/PostTest.php]: We need this since there is some leftover data that is used in the other tests. This way the tests will not fail.

**Review by sprankhub — APPROVED** [review]
